components/ILIAS/ResourceStorage/classes/Container/View/Configuration.php

```diff
@@ -74,6 +78,13 @@ public function withExternalAction(
         return $this;
     }
 
+    public function withExternalTopAction(string $key, TopAction $topAction, ?RoundTrip $modal): self
```

Comment:
The new parameter should be `?RoundTrip $modal = null`, I will open a PR with this.